diff --git a/autogen.sh b/autogen.sh
index fb34263..6751e28 100755
--- a/autogen.sh
+++ b/autogen.sh
@@ -447,6 +447,9 @@ if test $skip_gnulib = false; then
   $GNULIB_TOOL --lib=libunistring --source-base=lib --m4-base=gnulib-m4 --tests-base=tests \
     --with-tests --lgpl=3orGPLv2 --makefile-name=Makefile.gnulib --libtool --local-dir=gnulib-local \
     --avoid=float-h-tests \
+    --avoid=memchr-tests \
+    --avoid=strncat-tests \
+    --avoid=strncpy-tests \
     --import $GNULIB_MODULES || exit $?
   # Change lib/unistr.h to be usable standalone.
   sed -e 's/if GNULIB_[A-Za-z0-9_]* || .*/if 1/g' \
